Cabal/src/Distribution/Simple/Compiler.hs

```diff
@@ -329,6 +354,12 @@ data DebugInfoLevel
 instance Binary DebugInfoLevel
 instance Structured DebugInfoLevel
 
+instance Parsec DebugInfoLevel where
+  parsec = parsecDebugInfoLevel
+
+parsecDebugInfoLevel :: CabalParsing m => m DebugInfoLevel
+parsecDebugInfoLevel = flagToDebugInfoLevel <$> pure <$> parsecToken
```

Comment:
I don't have the typechecker at hand but could the left <$> be just `.`? I assume pure composition is better optimized, and the parser is somewhat performance-sensitive. That's a very minor concern of course.

Reply:
Works flawlessly, thanks (https://github.com/haskell/cabal/pull/8889/files#diff-1b1d4d646bf0cab8578f54a6a838a342c7a5787ea7469e2cf195a8959fc5a122R365)